### PR TITLE
ranking: Update docs

### PR DESCRIPTION
**Pull request by efritz**
Update user-facing documentation for background ranking jobs.

## Test plan

Docs update :)

### CODE REVIEW

File: doc/dev/background-information/architecture/precise-ranking.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+- A _SCIP exporter_, which maintains a _view_ of the relevant SCIP data for analysis, and
+- A _map/reduce_-like job that counts the number of references to a file across all indexed code.
+
+When the _exporter_ job first starts, all SCIP indexes will need to be initially exported. Once most visible indexes have been initially exported, the exporter will make incremental adjustments to this view to ensure it always reflects reality. SCIP indexes that are no longer visible are removed from this view, and new processed SCIP indexes becoming visible will be added.
+
+The _map-reduce_-like job starts, it will use whatever data is currently visible in the exported view. Until the initial export has completed, there will be ranking scores missing for some repositories with SCIP indexes once the calculation completes. The **cron expression** can be supplied to control how frequently and when to begin a new ranking calculation. This job is performed periodically, and its cadence can be controlled by the **cron expression** setting. This setting's value is parsed by Hashicorp's [`cronexpr`](https://github.com/hashicorp/cronexpr) library. Some sample valid expressions include:
```
Comment (jtibshirani) [inline doc/dev/background-information/architecture/precise-ranking.md:24]: The map-reduce-like job -> When the map-reduce-like job.

**Review by jtibshirani — APPROVED** [review]
Looks good to me, just left some small comments. I pushed an update to make it clear the code intel ranks are used as a big ranking signal. They don't affect the order of files on disk, so I reverted changes to that section.

File: doc/dev/background-information/architecture/precise-ranking.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Precise ranking
```
[the remark this answers is not in the document]
Comment (efritz) [reply doc/dev/background-information/architecture/precise-ranking.md:1]: 😎 , will update

[the remark this answers is not in the document]
Comment (efritz) [reply doc/dev/background-information/architecture/precise-ranking.md:1]: e8646f71584b

File: doc/dev/background-information/architecture/precise-ranking.md (hunk 1 above)
Comment (efritz) [reply doc/dev/background-information/architecture/precise-ranking.md:24]: 3c69af0463b6

**Pull request merged**